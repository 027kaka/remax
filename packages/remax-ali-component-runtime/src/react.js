export {
  Children,
  createRef,
  Component,
  PureComponent,
  createContext,
  forwardRef,
  lazy,
  memo,
  useCallback,
  useContext,
  useEffect,
  useImperativeHandle,
  useDebugValue,
  useLayoutEffect,
  useMemo,
  useReducer,
  useRef,
  useState,
  Fragment,
  Profiler,
  StrictMode,
  Suspense,
  unstable_SuspenseList,
  createElement,
  cloneElement,
  createFactory,
  isValidElement,
  version,
  unstable_withSuspenseConfig,
} from 'react';
