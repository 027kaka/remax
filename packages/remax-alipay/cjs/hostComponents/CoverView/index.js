'use strict';
var __importDefault =
  (this && this.__importDefault) ||
  function(mod) {
    return mod && mod.__esModule ? mod : { default: mod };
  };
Object.defineProperty(exports, '__esModule', { value: true });
var createHostComponent_1 = __importDefault(require('../../createHostComponent'));
var CoverView = createHostComponent_1.default('cover-view');
exports.default = CoverView;
