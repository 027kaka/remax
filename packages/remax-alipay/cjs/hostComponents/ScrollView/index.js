'use strict';
var __importDefault =
  (this && this.__importDefault) ||
  function(mod) {
    return mod && mod.__esModule ? mod : { default: mod };
  };
Object.defineProperty(exports, '__esModule', { value: true });
var createHostComponent_1 = __importDefault(require('../../createHostComponent'));
var ScrollView = createHostComponent_1.default('scroll-view');
exports.default = ScrollView;
