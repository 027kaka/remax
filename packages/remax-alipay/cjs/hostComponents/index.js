'use strict';
function __export(m) {
  for (var p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
}
Object.defineProperty(exports, '__esModule', { value: true });
var View_1 = require('./View');
exports.View = View_1.default;
__export(require('./View'));
var ScrollView_1 = require('./ScrollView');
exports.ScrollView = ScrollView_1.default;
__export(require('./ScrollView'));
var Swiper_1 = require('./Swiper');
exports.Swiper = Swiper_1.default;
__export(require('./Swiper'));
var SwiperItem_1 = require('./SwiperItem');
exports.SwiperItem = SwiperItem_1.default;
__export(require('./SwiperItem'));
var MovableView_1 = require('./MovableView');
exports.MovableView = MovableView_1.default;
__export(require('./MovableView'));
var MovableArea_1 = require('./MovableArea');
exports.MovableArea = MovableArea_1.default;
__export(require('./MovableArea'));
var CoverView_1 = require('./CoverView');
exports.CoverView = CoverView_1.default;
__export(require('./CoverView'));
var CoverImage_1 = require('./CoverImage');
exports.CoverImage = CoverImage_1.default;
__export(require('./CoverImage'));
var Icon_1 = require('./Icon');
exports.Icon = Icon_1.default;
__export(require('./Icon'));
var Text_1 = require('./Text');
exports.Text = Text_1.default;
__export(require('./Text'));
var RichText_1 = require('./RichText');
exports.RichText = RichText_1.default;
__export(require('./RichText'));
var Progress_1 = require('./Progress');
exports.Progress = Progress_1.default;
__export(require('./Progress'));
var Button_1 = require('./Button');
exports.Button = Button_1.default;
__export(require('./Button'));
var CheckboxGroup_1 = require('./CheckboxGroup');
exports.CheckboxGroup = CheckboxGroup_1.default;
__export(require('./CheckboxGroup'));
var Checkbox_1 = require('./Checkbox');
exports.Checkbox = Checkbox_1.default;
__export(require('./Checkbox'));
var Form_1 = require('./Form');
exports.Form = Form_1.default;
__export(require('./Form'));
var Input_1 = require('./Input');
exports.Input = Input_1.default;
__export(require('./Input'));
var Label_1 = require('./Label');
exports.Label = Label_1.default;
__export(require('./Label'));
var Picker_1 = require('./Picker');
exports.Picker = Picker_1.default;
__export(require('./Picker'));
var PickerView_1 = require('./PickerView');
exports.PickerView = PickerView_1.default;
__export(require('./PickerView'));
var PickerViewColumn_1 = require('./PickerViewColumn');
exports.PickerViewColumn = PickerViewColumn_1.default;
__export(require('./PickerViewColumn'));
var RadioGroup_1 = require('./RadioGroup');
exports.RadioGroup = RadioGroup_1.default;
__export(require('./RadioGroup'));
var Radio_1 = require('./Radio');
exports.Radio = Radio_1.default;
__export(require('./Radio'));
var Slider_1 = require('./Slider');
exports.Slider = Slider_1.default;
__export(require('./Slider'));
var Switch_1 = require('./Switch');
exports.Switch = Switch_1.default;
__export(require('./Switch'));
var Textarea_1 = require('./Textarea');
exports.Textarea = Textarea_1.default;
__export(require('./Textarea'));
var Navigator_1 = require('./Navigator');
exports.Navigator = Navigator_1.default;
__export(require('./Navigator'));
var Image_1 = require('./Image');
exports.Image = Image_1.default;
__export(require('./Image'));
var Map_1 = require('./Map');
exports.Map = Map_1.default;
__export(require('./Map'));
var Canvas_1 = require('./Canvas');
exports.Canvas = Canvas_1.default;
__export(require('./Canvas'));
var WebView_1 = require('./WebView');
exports.WebView = WebView_1.default;
__export(require('./WebView'));
var Lifestyle_1 = require('./Lifestyle');
exports.Lifestyle = Lifestyle_1.default;
__export(require('./Lifestyle'));
var ContactButton_1 = require('./ContactButton');
exports.ContactButton = ContactButton_1.default;
__export(require('./ContactButton'));
var Video_1 = require('./Video');
exports.Video = Video_1.default;
__export(require('./Video'));
