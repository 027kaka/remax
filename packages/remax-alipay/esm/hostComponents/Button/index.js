import createHostComponent from '../../createHostComponent';
var Button = createHostComponent('button');
export default Button;
