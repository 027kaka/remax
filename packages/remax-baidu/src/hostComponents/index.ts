export * from './View';
export * from './Button';
export * from './ScrollView';
export * from './Swiper';
export * from './SwiperItem';
export * from './Icon';
export * from './RichText';
export * from './Text';
export * from './Checkbox';
export * from './CheckboxGroup';
export * from './Radio';
export * from './RadioGroup';
export * from './Form';
export * from './Input';
export * from './Image';
export * from './Label';
export * from './Textarea';
export * from './Picker';
export * from './PickerView';
export * from './PickerViewColumn';
export * from './WebView';
