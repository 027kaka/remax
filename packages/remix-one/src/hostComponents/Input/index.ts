import * as React from 'react';
import { InputEvent } from '../../types';
import { createCallback, createInputEvent } from '../../createHostComponent';
import { InputProps } from './props';

export type { InputProps };

export interface InputState {
  value?: string;
  controlled: boolean;
}

export default class Input extends React.Component<InputProps, InputState> {
  state: InputState = {
    value: '',
    // 阿里
    controlled: false,
  };

  constructor(props: InputProps) {
    super(props);

    const controlled = props.value !== undefined;
    const value = controlled ? props.value : props.defaultValue;

    this.state = {
      value,
      controlled,
    };
  }

  componentDidUpdate() {
    if (this.props.value !== undefined && this.props.value !== this.state.value) {
      this.setState({ value: this.props.value });
    }
  }

  handleInput = (e: InputEvent) => {
    const { controlled } = this.state;

    if (!controlled) {
      this.setState({ value: e.target.value });
    }

    if (typeof this.props.onInput === 'function') {
      return this.props.onInput(e);
    }
  };

  render() {
    const inputProps = {
      ...this.props,
      onInput: createCallback(this.handleInput, createInputEvent),
    };

    if (inputProps.onConfirm) {
      inputProps.onConfirm = createCallback(this.props.onConfirm, createInputEvent);
    }

    if (inputProps.onFocus) {
      inputProps.onFocus = createCallback(this.props.onFocus, createInputEvent);
    }

    if (inputProps.onBlur) {
      inputProps.onBlur = createCallback(this.props.onBlur, createInputEvent);
    }

    return React.createElement('input', {
      ...inputProps,
      ...this.state,
    });
  }
}
