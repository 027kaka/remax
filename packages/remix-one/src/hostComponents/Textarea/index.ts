import * as React from 'react';
import { InputEvent } from '../../types';
import { createCallback, createInputEvent } from '../../createHostComponent';
import { TextareaProps } from './props';

export type { TextareaProps };

export interface TextareaState {
  value?: string;
  controlled: boolean;
}

export default class Textarea extends React.Component<TextareaProps, TextareaState> {
  state: TextareaState = {
    value: '',
    // 阿里
    controlled: false,
  };

  constructor(props: TextareaProps) {
    super(props);

    const controlled = props.value !== undefined;
    const value = controlled ? props.value : props.defaultValue;

    this.state = {
      value,
      controlled,
    };
  }

  componentDidUpdate() {
    if (this.props.value !== undefined && this.props.value !== this.state.value) {
      this.setState({ value: this.props.value });
    }
  }

  handleInput = (e: InputEvent) => {
    const { controlled } = this.state;

    if (!controlled) {
      this.setState({ value: e.target.value });
    }

    if (typeof this.props.onInput === 'function') {
      return this.props.onInput(e);
    }
  };

  render() {
    const inputProps = {
      ...this.props,
      onInput: createCallback(this.handleInput, createInputEvent),
    };

    if (inputProps.onConfirm) {
      inputProps.onConfirm = createCallback(this.props.onConfirm, createInputEvent);
    }

    if (inputProps.onFocus) {
      inputProps.onFocus = createCallback(this.props.onFocus, createInputEvent);
    }

    if (inputProps.onBlur) {
      inputProps.onBlur = createCallback(this.props.onBlur, createInputEvent);
    }

    return React.createElement('textarea', {
      ...inputProps,
      ...this.state,
    });
  }
}
