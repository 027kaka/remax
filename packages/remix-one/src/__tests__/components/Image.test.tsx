import * as React from 'react';
import TestRenderer from 'react-test-renderer';
import { Image } from '../../hostComponents';

describe('Image', () => {
  it('render correctly in ali', () => {
    process.env.REMAX_PLATFORM = 'ali';
    const testRenderer = TestRenderer.create(
      <Image
        className="class"
        onError={() => {
          // ignore
        }}
        onLoad={() => {
          // ignore
        }}
      />
    );

    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
});
