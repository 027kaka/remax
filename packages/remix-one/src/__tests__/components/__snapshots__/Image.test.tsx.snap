// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image render correctly in ali 1`] = `
<image
  className="class"
  onError={[Function]}
  onLoad={[Function]}
/>
`;
